servo-core/src/main/java/com/netflix/servo/monitor/DynamicGauge.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2013-2018 Netflix, Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.servo.monitor;
+
+import com.google.common.base.Throwables;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.netflix.servo.DefaultMonitorRegistry;
+import com.netflix.servo.tag.TagList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class that dynamically creates gauges based on an arbitrary (name, tagList),
+ * or {@link com.netflix.servo.monitor.MonitorConfig}
+ * Gauges are automatically expired after 15 minutes of inactivity.
+ */
+public final class DynamicGauge implements CompositeMonitor<Long> {
```

Comment:
If it is not recommended for general use, do you think we should mark it deprecated to warn users?

Reply:
In my mind, we should mark all of servo as `@Deprecated` and have people migrate to spectator. For this particular use case, we don't have a way within servo to do the same. I'd have to point them to `PolledMeter` or active gauges in spectator.